### PR TITLE
Dave and Bambi engine (WIP)

### PR DESCRIPTION
**Pull request by Wither362**
Heres for the dave and bambi lovers!

heres also your engine @PHGaming121

### CODE REVIEW

File: source/PlayState.hx [hunk 1]
```diff
@@ -1228,6 +1256,53 @@ class PlayState extends MusicBeatState
 		scoreTxt.visible = !ClientPrefs.noscore;
 		add(scoreTxt);
 
+		if(SONG.credit != null) {
+			watermarkTxt = new FlxText(10, FlxG.height - 28, 0, 'By ' + SONG.credit + ' - ' + SONG.song + " - [" + CoolUtil.difficultyString() + "] - Theoyeah Engine", 74);
```
Comment (AverageDaveLover65) [inline source/PlayState.hx:1258]: what the fuck i just noticed, this is from strident wtfffff

File: source/PlayState.hx [hunk 2]
```diff
@@ -1228,6 +1256,53 @@ class PlayState extends MusicBeatState
...
+			watermarkTxt.setFormat(Paths.font("vcr.ttf"), 16, FlxColor.WHITE, LEFT, FlxTextBorderStyle.OUTLINE, FlxColor.BLACK);
+			watermarkTxt.size = 18;
+			watermarkTxt.updateHitbox();
+			watermarkTxt.alpha = 0.6;
+			add(watermarkTxt);
+		}
+
+		if(SONG.screwYou != null)
+		{
+			screwYouTxt = new FlxText(10, FlxG.height - 28, 0, SONG.screwYou, 74);
```
Comment (AverageDaveLover65) [inline source/PlayState.hx:1278]: THIS IS ALL FROM STRIDENT WTFFFF

File: source/PlayState.hx [hunk 3]
```diff
@@ -2994,11 +3070,34 @@ class PlayState extends MusicBeatState
...
+		if(activeWavy)
+		{
+			wavyShader.update(elapsed);
+		}
+		the3DWorldEffect.update(elapsed);
+		the3DWorldEffectWavy.update(elapsed);
+		
+		if(disableTheTripperAt == curStep)
+		{
+			disableTheTripper = true;
```
[the remark this answers is not in the document]
Comment (Wither362) [reply source/PlayState.hx:3082]: I now it :trollface:

**Comment by Jevilm** [comment]
do compiled versoni!!!!!!!!

**Comment by Wither362** [comment]
> do compiled versoni!!!!!!!!

its broken, no compiled version until i repair it

**Comment by PHGaming121** [comment]
What is broken @Wither362 ?

**Comment by PHGaming121** [comment]
If the code of the wavy BG and the rainbow eyesores event the try to add the code from here:

https://github.com/impostor5875/the-funny/tree/the-manny-men

This version already have this things

**Comment by Wither362** [comment]
> If the code of the wavy BG and the rainbow eyesores event then try to add the code from here:
> 
> https://github.com/impostor5875/the-funny/tree/the-manny-men
> 
> This version already have this things

Ok, seems like that code is right… I will copy it and edit it tomorrow :trollface:

**Comment by Wither362** [comment]
Oh, and don't worry about I copying code, I will also create mine without copying

**Comment by AverageDaveLover65** [comment]
> > If the code of the wavy BG and the rainbow eyesores event then try to add the code from here:
> > https://github.com/impostor5875/the-funny/tree/the-manny-men
> > This version already have this things
> 
> Ok, seems like that code is right… I will copy it and edit it tomorrow :trollface:

so you admittedly said you stole the code, what the fuck is wrong with you

**Comment by Wither362** [comment]
> > > If the code of the wavy BG and the rainbow eyesores event then try to add the code from here:
> > > https://github.com/impostor5875/the-funny/tree/the-manny-men
> > > This version already have this things
> > 
> > Ok, seems like that code is right… I will copy it and edit it tomorrow :trollface:
> 
> so you admittedly said you stole the code, what the fuck is wrong with you

Oh yeah?? IM A FUCKING 13 YEARS OLD KID!!! AND A KID WHO KNOWS HOW TO CODE IS A LITTLE BIT CONFUSING RIGHT!?!?

**Comment by Wither362** [comment]
Oh, and I will not stole it, I will make their credits

**Comment by Wither362** [comment]
Ok, prepare yourself for Dave and Bambi because today I will fix errors, add more events, and more!

**Comment by PHGaming121** [comment]
How to fix this?
186904197-fff028b5-b57e-4ea9-a109-2364f582aac0

**Comment by Wither362** [comment]
> How to fix this?
> 186904197-fff028b5-b57e-4ea9-a109-2364f582aac0

AHGAJAJGJAJFIAOJDIDOAKA!!!!

**Comment by PHGaming121** [comment]
@Theoyeah can u help?

**Comment by Wither362** [comment]
> OK @Wither362 the engine finally opening but.......
> NEW ERROR WHEN ENTER SONG
> Captura de Tela (208)

Ok, that is fixed now

**Comment by PHGaming121** [comment]
NOT AGAAAAAAAAAAIN

**Comment by PHGaming121** [comment]
engine not open, just said "null object reference"

**Comment by PHGaming121** [comment]
no
didnt working AAAAAAAAAAAAAAAAA

**Comment by PHGaming121** [comment]
@Wither362 IS THIS WORKING??????

**Comment by Wither362** [comment]
> @Wither362 IS THIS WORKING??????

Wtf I’m at school, I can’t see it right now

**Comment by Wither362** [comment]
> ALL OF THIS IS FROM STRIDENT WTFF

Ok, yes, I know it, but that doesn’t mean I can’t take the code right? I have credit Joalor, so I don’t think there is any problem, now I have a question, do you want to use this pull request?

**Comment by PHGaming121** [comment]
188509613-18554acb-12a4-4543-b3db-4115a4f25da0

SDHFOSAHOHVUHSJOHBVFBOUNSVASIUBFSHAUVBFSADGVAVGUBVYFGYSVAGUFHVGAUYGFAFSA

WHYYYYYYYYYYYYYYYYY

**Comment by Wither362** [comment]
Ok, @PHGaming121, looks like I’ve to do some things to repair it, so, relax and I will try to fix it by using other code

**Comment by Wither362** [comment]
This should be repaired right?

**Comment by PHGaming121** [comment]
> This should be repaired right?

IDK

**Comment by MemeHoovy** [comment]
> This should be repaired right?

idk, you asked yourself

**Comment by Wither362** [comment]
Ok, this will be closed, I will continue it [here: https://github.com/Wither362/Theoyeah-Engine-DaveAndBambi](https://github.com/Wither362/Theoyeah-Engine-DaveAndBambi)

**Pull request closed without merging**